airflow/models/xcom.py

```diff
@@ -124,85 +122,71 @@ def set(
 
     @classmethod
     @provide_session
-    def get_one(cls,
-                execution_date,
-                key=None,
-                task_id=None,
-                dag_id=None,
-                include_prior_dates=False,
-                session=None):
+    def compose_query(cls,
```

Comment:
I think you could leave this called `get_many` -- although we used to do this

```
        results = query.all()
        return results
```

The `.all()` call is optional, and SQLAlchemy will correctly and happily allow you to iterate over a query.

```
    @classmethod
    def get_many(...):
        return query
```

And then you could use it either like `for xcom in XCom.get_many(...):` or `XCom.get_many().filter(extra='condition')`

Reply:
That's actually a good point. I didn't know that the `.all()` is optional, thanks!